pandas/core/groupby.py

```diff
@@ -2368,11 +2369,21 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
     # axis of the object
     if level is not None:
         if not isinstance(group_axis, MultiIndex):
+            if is_list_like(level):
```

Comment:
add a comment here *why* this is needed.

you are setting ``level``, but below we reset to ``None`` (see L2389), so something wrong.